src/app/decorations/qgsdecorationtitledialog.cpp

```diff
@@ -0,0 +1,136 @@
+/***************************************************************************
+  qgsdecorationtitledialog.cpp
+  --------------------------------------
+  Date                 : November 2018
+  Copyright            : (C) 2018 by Mathieu Pellerin
+  Email                : nirvn dot asia at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsdecorationtitledialog.h"
+#include "qgsdecorationtitle.h"
+
+#include "qgisapp.h"
+#include "qgsexpression.h"
+#include "qgsexpressionbuilderdialog.h"
+#include "qgsexpressioncontext.h"
+#include "qgshelp.h"
+#include "qgsmapcanvas.h"
+#include "qgssettings.h"
+
+#include <QColorDialog>
+#include <QColor>
+#include <QFont>
+#include <QDialogButtonBox>
+#include <QPushButton>
+
+QgsDecorationTitleDialog::QgsDecorationTitleDialog( QgsDecorationTitle &deco, QWidget *parent )
+  : QDialog( parent )
+  , mDeco( deco )
+{
+  setupUi( this );
+  connect( buttonBox, &QDialogButtonBox::accepted, this, &QgsDecorationTitleDialog::buttonBox_accepted );
+  connect( buttonBox, &QDialogButtonBox::rejected, this, &QgsDecorationTitleDialog::buttonBox_rejected );
+  connect( mInsertExpressionButton, &QPushButton::clicked, this, &QgsDecorationTitleDialog::mInsertExpressionButton_clicked );
+
+  QgsSettings settings;
```

Comment:
Use

   QgsGui::enableAutoGeometryRestore( this );

instead of the manual approach